client/components/ChallengesList.js

```diff
@@ -26,12 +33,45 @@ const component = ({
   useEffect(() => {
     if (difficulty) {
       filter = difficulty;
+      fetchAllUserchallenges();
       fetchChallengesWithFilterAndSearch(filter, searchTerm);
     } else {
       fetchChallenges();
+      fetchAllUserchallenges();
     }
   }, [difficulty, searchTerm]);
 
+  // console.log(typeof userchallenge)
+  const solutionBychallengeId = challengeId => {
+    if (userchallenge) {
+      const solutions = userchallenge.filter(solution => solution.challengeId === challengeId);
+      return solutions;
+    }
+  };
+
+  const attemptedTimes = arr => {
+    return arr.length;
+  };
+
+  const attemptedByUsers = arr => {
+    const userId = [];
```

Comment:
Make this plural indicating this can have multiple ids.  Also change for the next few lines.
```suggestion
    const userIds = [];
```